packages/ui/src/Alert.tsx

```diff
@@ -43,21 +43,23 @@ export type AlertProps = {
   content: ReactNode
   actions?: AlertAction[]
   className?: string
+  dismissableByEscKey?: boolean
   closeToast?: (e?: React.MouseEvent<HTMLButtonElement>) => void
 }
 
-export const Alert: FC<AlertProps> = ({ type, title, content, actions, className, closeToast }) => {
+export const Alert: FC<AlertProps> = ({ type, title, content, actions, className, dismissableByEscKey = true, closeToast }) => {
   const { icon, ariaLabel } = ToastIcon[type]
 
-  useKey(
-    escKeyFilterPredicate,
+  const closeByEsc = useCallback(
     (nativeEvent: KeyboardEvent) => {
-      closeToast && closeToast()
+      dismissableByEscKey && closeToast?.()
       nativeEvent.stopImmediatePropagation()
```

Comment:
@utkukaratas should we stopImmediatePropagation if the alert is not dismissable by esc?

Reply:
aaaaaaah! you've one hell of an eye for code andrey!!! fix incoming...